paramiko/common.py

```diff
@@ -160,16 +160,18 @@
 
 
 def asbytes(s):
-    if not isinstance(s, bytes_types):
-        if isinstance(s, string_types):
-            s = b(s)
-        else:
-            try:
-                s = s.asbytes()
-            except Exception:
-                raise Exception('Unknown type')
+    if isinstance(s, bytes_types):
+        return s
+    if isinstance(s, text_type):
+        # GZ 2017-05-25: Accept text and encode as utf-8 for compatibilty only.
```

Comment:
Was wondering that myself. Was wondering if some sort of joke around the submitter's Github username (`bz2`) :D